Update-metodi tapahtumille

nyDropTheC opened the pull request:
Näyttää ekalta katseelta OKlta, mutta kaipaa tarkastusta

nyDropTheC committed:
Merge branch 'develop' into rest-api-events

nyDropTheC committed:
Merge branch 'develop' into rest-api-events

nyDropTheC committed:
Merge branch 'develop' into rest-api-events

nyDropTheC committed:
Implemented event updates

Implemented updates and added the method to API documentation.
Is it OK?

Alttz commented:
Mä sanoisin, et kyllä toi toimii, mut ei ehkä oo paras tapa toteuttaa niin, että ei endpointissa määritetä id:tä jo. Eli nyt tolla pystyy muokkaamaan käytännössä kaikkia tapahtumia kerralla, joka ei ehkä oo tarkoituksen mukaista.

Mä menisin kuitenki tällä:

@PutMapping("/api/event/{id}")
    public ResponseEntity<Event> updateEvent(@PathVariable Long id, @RequestBody Event updatedEvent) {
        if (!erepository.existsById(id)) {
            return ResponseEntity.notFound().build();
        }
        updatedEvent.setId(id); 
        return ResponseEntity.ok(erepository.save(updatedEvent));
    }

nyDropTheC committed:
Made updateEvent more RESTlike

- Added an id path variable as outlined in pull req
- Updated documentation to reflect new state of events
